msms recipe

rhs2132 opened the pull request:
This is a simple recipe for installing msms. A similar package exists on bioconda but intentionally excludes windows compatibility. Please let me know if I need to make any changes.

Ryan

<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
or on our [Keybase chat](https://keybase.io/team/condaforge.chat)
if your recipe isn't reviewed promptly.
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/msms-exe```) and found some lint.

Here's what I've got...


For **recipes/msms-exe**:

 * There are too few lines.  There should be one empty line at the end of the file.
 * The extra section contained an unexpected subsection name. skip-lints is not a valid subsection name.

For **recipes/msms-exe**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/msms-exe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/msms-exe**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

rhs2132 commented:
Thanks in advance for the help getting this to work. I have been trying to test the recipe locally as recommended using the .circleci/run_docker_build.sh for linux64. Everything runs seemingly correctly but exists with the message "Found no recipes to build". I think that I am missing something obvious but can't figure out what it is. I have attached the stdio output from the docker script.


[docker_log.txt](https://github.com/conda-forge/staged-recipes/files/4743308/docker_log.txt)

rhs2132 commented:
I fixed the previous issue: my repository was pointing to my fork so my recipe was getting automatically removed from the recipe folder. I added the build requirement of the c compiler ({{ compiler('c') }}) to the recipe which helped a little. I am now getting an overlinking error. I have been searching the forums but have not found a fix yet.
[docker_log.txt](https://github.com/conda-forge/staged-recipes/files/4747348/docker_log.txt)
 I have pasted the specific warnings/errors below with the full stdio output attached:

WARNING :: get_rpaths_raw()=[] and patchelf=[['']] disagree for /home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/msms :: 
WARNING: Disagreement in get_linkages(filename=/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/msms, resolve_filenames=True, recurse=False, sysroot=/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/, envroot=/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl, arch=native):
 lief: {'/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6', '/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6'}
pyldd: {'/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/lib/libc.so.6', '/home/conda/staged-recipes/build_artifacts/msms-exe_1591633884994/_build_env/x86_64-conda-linux-gnu/sysroot/lib/libm.so.6'}
  (using lief)
  ERROR (msms-exe,bin/msms): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 not found in any CDT/compiler package, nor the whitelist?!
  ERROR (msms-exe,bin/msms): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 not found in any CDT/compiler package, nor the whitelist?!
Traceback (most recent call last):
  File "/home/conda/.ci_support/build_all.py", line 134, in <module>
    build_all(args.recipes_dir, args.arch)
  File "/home/conda/.ci_support/build_all.py", line 67, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File "/home/conda/.ci_support/build_all.py", line 124, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2864, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2151, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1505, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1364, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1178, in post_build
    check_overlinking(m, files, host_prefix)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1110, in check_overlinking
    m.config.enable_static)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/post.py", line 1072, in check_overlinking_impl
    raise OverLinkingError(overlinking_errors)
conda_build.exceptions.OverLinkingError: overlinking check failed 
['  ERROR (msms-exe,bin/msms): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 not found in any CDT/compiler package, nor the whitelist?!', '  ERROR (msms-exe,bin/msms): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 not found in any CDT/compiler package, nor the whitelist?!']

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/msms-exe```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/msms-exe```) and found some lint.

Here's what I've got...


For **recipes/msms-exe**:

 * Failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint .` from the recipe directory.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/msms-exe```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/bio-msms```) and found it was in an excellent condition.

rhs2132 commented:
@conda-forge/staged-recipes This is ready for review/merge. This is modeled after the bioconda msms package but with added windows compatibility. The license is custom. I pulled the relevant file into the packaged LICENSE.txt and classified it as AFL-3.0 after comparing the license with the SPDX options. Let me know if I need to make any more changes. Thanks!

scopatz commented:
> The name change to bio-msms is to avoid confusion with the bioconda package

We should probably have consistent naming

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/bio-msms```, ```recipes/msms```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/msms**:

 * Recipe with the same name exists in bioconda: please discuss with @conda-forge/bioconda-recipes.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/msms```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/msms**:

 * Recipe with the same name exists in bioconda: please discuss with @conda-forge/bioconda-recipes.

rhs2132 commented:
I changed the name to msms. You're right that if the packages are functionally identical on systems where they are both available that they should have the same name.

@conda-forge/bioconda-recipes What do you think the proper name should be? I have been using the bioconda msms package but want to deploy some code with the same dependency onto windows. I have not replicated the pdb_to_xyzr and pdb_to_xyzrn scripts in this package because identical functionality is available through biopython albeit through a private method. I can reincorporate this functionality for unix pretty easily so that it is a direct, drop-in replacement for the bioconda package on unix systems.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/bio-pyvol```, ```recipes/msms```) and found some lint.

Here's what I've got...


For **recipes/bio-pyvol**:

 * The `requirements/` sections should be defined in the following order: build, host, run; instead saw: host, build, run.
 * When defining a source/url please add a sha256, sha1 or md5 checksum (sha256 preferably).
 * license_file entry is missing, but is required.

For **recipes/msms**:

 * Recipe with the same name exists in bioconda: please discuss with @conda-forge/bioconda-recipes.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/bio-pyvol```, ```recipes/msms```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/msms**:

 * Recipe with the same name exists in bioconda: please discuss with @conda-forge/bioconda-recipes.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/bio-pyvol```, ```recipes/msms```) and found some lint.

Here's what I've got...


For **recipes/bio-pyvol**:

 * license_file entry is missing, but is required.

For **recipes/msms**:

 * Recipe with the same name exists in bioconda: please discuss with @conda-forge/bioconda-recipes.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/bio-pyvol```, ```recipes/msms```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/msms**:

 * Recipe with the same name exists in bioconda: please discuss with @conda-forge/bioconda-recipes.

scopatz commented on recipes/bio-pyvol/meta.yaml:
```diff
@@ -0,0 +1,61 @@
+    - python
+    - scikit-learn >=0.20.2
+    - scipy >=1.2.1
+    - trimesh >=2.36.29
```
Are most of these needed for building the package?

rhs2132 replied:
I just did a more thorough re-read of the descriptions for the build, host, and run sections. I'm cleaning this up.

scopatz commented on recipes/msms/meta.yaml:
```diff
@@ -0,0 +1,37 @@
+  - url: {{ url_base }}/REL{{ version }}/msms_MacOSX_{{ version }}.tar.gz  # [osx]
+    md5: 36bd8447908d12a8703f458564cbe74b  # [osx]
+  - url: {{ url_base }}/REL{{ version }}/msms_win32_{{ version }}.zip  # [win]
+    md5: 482c38ca00a2c091cdc0807a8a272386  # [win32]
```
Please build from source

rhs2132 replied:
What does that mean exactly? Scrips has only released this code as precompiled executables. I have modeled my recipe after the one in bioconda: 
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/msms/meta.yaml

scopatz replied:
probably worth asking them

rhs2132 replied:
I have reached out to the authors and maintainers at Scripps (through MGL Tools) several times over the last year, but I have yet to hear back. I have also checked distribution for tools that rely on this executable. CCP4, VMD, Biopython, and PyMOL all can check for the presence of MSMS and read its output but do not distribute the executable itself. I would be comfortable using the Bioconda channel's version except that it excludes Windows compatibility.

What steps can I take to move this forward?

scopatz replied:
Yeah this seems to be the only way then.  Can you please use sha256 for your hashes here, though.  I think the various bots might break if you use md5
